src/server/events/index.ts

```diff
@@ -0,0 +1,114 @@
+import ical, { CalendarResponse, CalendarComponent, VEvent } from 'node-ical';
+import { AuthenticationError, UserInputError } from 'apollo-server-express';
+import { ObjectID } from 'mongodb';
+import { EventUpdateInput, EventDbObject } from '../generated/graphql';
+import { Models } from '../models';
+import { EventUpdate } from '../../client/routes/events/ManageEventTypes';
+
+const EVENTTYPE = 'VEVENT';
+
+const filterByCalType = (objNames: string[], cal: CalendarResponse, type: string): string[] => {
+	return objNames.filter(obj => cal[obj].type === type);
+};
+
+const filterCalByObjectNames = (objNames: string[], cal: CalendarResponse): CalendarComponent[] => {
+	return objNames.map(key => cal[key]);
```

Comment:
why is this method called filter if it's actually a map? rename func?

Reply:
It uses map but functionally it is returning only entries in `cal` using the list of objNames as a filter

Reply:
it is possible objNames is not in cal right? in that case, handle error? probably shouldn't return a bunch of `undefined` in the final array

Reply:
ObjNames is derived from the calendar; it is strict subset of keys in the calendar

Reply:
wdym derived? it is possibly someone could call that func with any string arr and mess it up...

Reply:
This func is limited to the scope of events and is only called on CalendarResponse type objs, why would someone use this func without context of what objnames actually is

Reply:
ok fine